### ISSUE TITLE
Docs

### ISSUE THREAD

**Issue by willtebbutt** [SciML/Integrals.jl#32]
Firstly, thanks for making this fantastic package! It's great to have the various quadrature / cubature packages accessible via a common interface.

The docs are currently a little harder to parse than they could be though. In particular:

1. The first example in the readme is kind of hard to parse because it's not explained what each of the arguments does until further down in the README. It would be helpful if there was at least quickly suggested that the first argument is the function you want to integrate, the second lower bound(s), and the third upper bound(s).
2. the second example is quite hard to parse, and I'm still not quite sure a) what format I can expect `x` to have and b) what format I can expect `dx` to have, as this doesn't appear to be documented. An explanation of precisely what I should assume about the structure of `x`, and `dx` (presumably as a function of `x`?) would be greatly appreciated.
3. assumptions about `f` when a parameter is not needed. It would be helpful to explicitly state that you still need to have an argument in `f` that is the parameters, assuming I've understood this correctly.
4. 1, 2 and 3 would be greatly appreciated in docstring-form, in conjunction with the information in the `QuadratureProblem` section of the README.
5. `solve` doesn't appear to be properly documented anywhere, although I'll grant you that it's pretty obvious how to use it from the example. Regardless, at least the presence of a docstring would be great.
6. It's unclear what to do with the output type of a successful `solve` call. I think the `u` field is the thing I'm after, but I don't really know? It looks like this is defined in `DiffEqBase`, so maybe the docs already exist for it? If so, it would be great if they could be linked. Again, maybe a docstring for the type?

Thanks again for the package!

**Comment by ChrisRackauckas** [comment]
All good suggestions. Quite busy right now but these should all be done.

**Comment by scheidan** [comment]
To add to the list above

7. What happens if kwargs are passed to `QuadratureProblem`? I think currently they are not used anywhere.

**Comment by ChrisRackauckas** [comment]
They should splat in like any other kwargs. It's not fully handled right now.

**Comment by ArnoStrouwen** [comment]
I think most of this is implemented now?
